src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import className from 'classnames';
+import { TaskType } from '../../types/TaskType';
+import { TodosContext } from '../TodoContext/TodoContext';
+
+type Props = {
+  todo: TaskType,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, title, completed } = todo;
+  const { addTodo } = useContext(TodosContext);
+  const [isEditable, setIsEditable] = useState(false);
+  const [editedTodo, setEditedTodo] = useState(title);
+
+  const fieldTitle = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (fieldTitle.current && isEditable) {
+      fieldTitle.current.focus();
+    }
+  }, [isEditable]);
+
+  const handleChangeStatus = () => {
+    addTodo(currentTodos => (
+      currentTodos.map(currentTodo => (
+        currentTodo.id === id
+          ? {
+            ...currentTodo,
+            completed: !currentTodo.completed,
+          }
+          : currentTodo
+      ))
+    ));
+  };
```

Comment:
I think it is better to create a separate method in context and in the component call it with particular todo as a parametr

Reply:
Fixed